src/sys/socket/mod.rs

```diff
@@ -69,23 +69,39 @@ pub enum SockType {
 
 /// Constants used in [`socket`](fn.socket.html) and [`socketpair`](fn.socketpair.html)
 /// to specify the protocol to use.
-#[repr(i32)]
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-pub enum SockProtocol {
-    /// TCP protocol ([ip(7)](http://man7.org/linux/man-pages/man7/ip.7.html))
-    Tcp = libc::IPPROTO_TCP,
-    /// UDP protocol ([ip(7)](http://man7.org/linux/man-pages/man7/ip.7.html))
-    Udp = libc::IPPROTO_UDP,
-    /// Allows applications and other KEXTs to be notified when certain kernel events occur
+pub enum SockProtocol {   
+    /// htons conversion for RAW type
+    Htons(u16),
```

Comment:
Since we are going to use plain numbers instead of fixed enums, why not make an entry like `Custom` that accept just an integer instead of `Htons`?
I mean a procotol number (in general) is just an arbitrary number, which (again, in general) doesn't have to do anything with the byte order, at least that is my understanding.

What'd you say?

Reply:
Thanks for reviewing the code!
Htons gives a conversion from `u16` to `htons(u16)`,
but yes, perhaps it might be best to also have a Integer(i32).

Two things:
1. `man socket(2)` correctly states that `The protocol number to use is specific to the communication domain in which communication is to take place.` meaning that, as you said, a protocol is just an arbitrary number for the sock type.
2. I like the idea of looking at API documentation and getting a feel for what is possible. That's what makes the enum so great.

So, how about keeping the existing enums, but adding an Integer(i32) enum type?